spanner_orm/decorator.py

```diff
@@ -33,33 +35,24 @@ def get_book(transaction, book_id):
     To call a decorated function if you already have a transaction:
 
     @transactional_read
-    def list_books(transaction, book_ids):
+    def list_books(book_ids, transaction=None):
       for book_id in book_ids:
         get_book(book_id, transaction=transaction)
 
     list_books method would call get_book method using same transaction
 
   Args:
     func: Callable which will be called with read-only transaction and original
-      arguments. `func` can also be passed an optional 'transaction' kwarg to
-      use a given transaction, instead of creating a new one. This means `func`
-      cannot define a keyword arg of its own called 'transaction'.
+      arguments. Decorated `func` can also be passed an optional 'transaction'
+      kwarg to use a given transaction, instead of creating a new one.
 
   Returns:
     decorated function
   """
+  return _transactional(api.SpannerApi.run_read_only, func)
 
-  def wrapper(*args, **kwargs) -> Any:
-    if 'transaction' in kwargs:
-      transaction = kwargs['transaction']
-      del kwargs['transaction']
-      return func(transaction, *args, **kwargs)
-    return api.SpannerApi.run_read_only(func, *args, **kwargs)
-
-  return wrapper
 
-
-def transactional_write(func: Callable[..., Any]) -> Callable[..., Any]:
+def transactional_write(func: Callable[..., T]) -> Callable[..., T]:
```

Comment:
Please update this function's docstring to match the new way `transaction` parameters work.

Reply:
And I see you just pushed a commit to do this.